onnxruntime/core/providers/cuda/tensor/sequence_op.h

```diff
@@ -10,16 +10,6 @@
 namespace onnxruntime {
 namespace cuda {
 
-template <typename DataType>
-int64_t ReadIndex(const Tensor& tensor, const char* type_name) {
-  DataType data{0};
-  if (!CUDA_CALL(cudaMemcpy(&data, tensor.Data<DataType>(),
```

Comment:
....When really the input is needed on CPU which means such logic needs to exist to bring it "back to CPU"